src/ng2-smart-table/components/filter/filter-types/checkbox-filter.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { DefaultFilter } from './default-filter';
+import { Column } from '../../../lib/data-set/column';
+
+@Component({
+  selector: 'checkbox-filter',
+  styles: [`
+    a {
+      font-weight: normal;
+    }
+  `],
+  template: `
+    <input type="checkbox" #filterEl [ngClass]="inputClass" class="form-control">
+    <a href="#" *ngIf="filterActive" (click)="resetFilter($event)">{{column.getFilterConfig()?.resetText || 'reset'}}</a>
+  `
+})
+export class CheckboxFilterComponent extends DefaultFilter implements OnInit {
+
+  filterActive: boolean = false;
+
+  constructor() {
+    super();
+  }
+
+  ngOnInit() {
+    this.changesSubscription = Observable.fromEvent(this.filterEl.nativeElement, 'change')
```

Comment:
I don't think that will work since the value of the checkbox does not change.. what do you think? Maybe I could check that simply with the `(chage)` event but I wanted to know why you would prefer that over the Observable pattern. Just for personal knowledge and for future reference.
Thanks